mongo/src/test/scala/io/sphere/mongo/format/OptionMongoFormatSpec.scala

```diff
@@ -0,0 +1,104 @@
+package io.sphere.mongo.format
+
+import io.sphere.mongo.generic._
+import org.scalatest.{MustMatchers, OptionValues, WordSpec}
+import io.sphere.mongo.MongoUtils._
+import DefaultMongoFormats._
+
+object OptionMongoFormatSpec {
+
+  case class SimpleClass(value1: String, value2: Int)
+
+  object SimpleClass {
+    implicit val mongo: MongoFormat[SimpleClass] = mongoProduct(apply _)
+  }
+
+  case class ComplexClass(
+    name: String,
+    simpleClass: Option[SimpleClass])
+
+  object ComplexClass {
+    implicit val mongo: MongoFormat[ComplexClass] = mongoProduct(apply _)
+  }
+
+}
+
+
+class OptionMongoFormatSpec extends WordSpec with MustMatchers with OptionValues {
+  import OptionMongoFormatSpec._
+
+  "MongoFormat[Option[_]]" should {
+    "handle presence of all fields" in {
+      val dbo = dbObj(
+        "value1" -> "a",
+        "value2" -> 45
+      )
+      val result = MongoFormat[Option[SimpleClass]].fromMongoValue(dbo)
+      result.value.value1 mustEqual "a"
+      result.value.value2 mustEqual 45
+    }
+
+    "handle presence of all fields mixed with ignored fields" in {
+      val dbo = dbObj(
+        "value1" -> "a",
+        "value2" -> 45,
+        "value3" -> "b"
+      )
+      val json =
```

Comment:
that json is not used, is it?

Reply:
👍 https://github.com/sphereio/sphere-scala-libs/pull/75/commits/9ba55024275bc32ed6cbe819c509964638c32f7c